catalog/tests/dags/common/test_ingestion_server.py

```diff
@@ -19,7 +55,82 @@
     ],
 )
 def test_response_filter_stat(data, expected):
-    response = MagicMock()
+    response = mock.MagicMock()
     response.json.return_value = data
     actual = ingestion_server.response_filter_stat(response)
     assert actual == expected
+
+
+@pytest.mark.parametrize(
+    "response_code, response_json, environment, expected_status",
+    [
+        # Production
+        # Healthy cluster
+        (200, {"status": "green"}, "prod", TaskInstanceState.SUCCESS),
```

Comment:
Just a nit/tip, you can give parametrize param groups a custom id. This helps make it easier to identify which ones fail in the test output as well because pytest uses the id rather than concatenating each param value's `repr` to name the parametrized instance.

```suggestion
        pytest.param(
        200,
        {"status": "green"},
        "prod",
        TaskInstanceState.SUCCESS,
        id="healthy-cluster"
       ),
```

If this test failed, the test name would be `test_index_readiness_check[healthy-cluster]` instead of `test_index_readiness_check[200-response_json0-prod-SUCCESS]`